hyperion/views/post_views.py

```diff
@@ -50,8 +50,16 @@ def get_auth_posts(self, request):
         '''
         GET /author/posts
         '''
-        serializer = PostSerializer(
-            self.queryset.filter(author=request.user.profile), many=True)
+
+        # result = list of post
+        result = list(self.queryset.filter(author=request.user.profile))  \
+        + Post.visible_to_private(request.user.profile) \
+        + Post.visible_to_public()  \
+        + Post.visible_to_friends_of_friends(request.user.profile)  \
+        + Post.visible_to_friends(request.user.profile)
```

Comment:
TBH, I don't like this approach. This will significantly increase the response time of the API.

Reply:
There aren't any trivial approach here. Since visiable_to only works when the visibility is set to private.  I have chosed this approach since the data size will be small in our case.